nncf/quantization/algorithms/weight_compression/algorithm.py

```diff
@@ -470,14 +479,14 @@ def apply(
                 backend_entity=self._backend_entity,
             )
         else:
-            if self._scale_estimation and activations is not None and self._mode != CompressWeightsMode.E2M1:
```

Comment:
Removed additional checks from this and similar conditions to avoid redundancy. Firstly, activations must always be present if data-aware compression is enabled. Secondly, the check for E2M1 is already present inside the `nncf.compress_weights()` call.